### PR TITLE
fix: interaction partners viewer height

### LINKED ISSUE #1176: Labels are offset after selecting a node for Interaction Partners
**Describe the bug**

The labels are offset when click selecting a node for Interaction Partners

**Steps to reproduce**

1. Open the IP for e.g. `ENSG00000144290`
2. Select any node
3. The labels will be offset 

**Expected behavior**

- [ ] Labels will not be offset when selecting a node 

**Screenshots**

#### After selecting

**Additional context**

After expanding the network, labels will not be offset when selecting a node

### PR DESCRIPTION
**Pull request by e0**
**Related issue(s) and PR(s)**  
This PR closes #1176 and #1207.

**Type of change**  

- [x] Bug fix (non-breaking change which fixes an issue)
 
**List of changes made**  

1. Set fixed viewer height after page load.
2. Fix mobile viewer height after page load. (This is and unreported bug, where viewer height is `0`.)
3. Add handler to adjust viewer height after window resize.

**Testing**  

- Relative urls that can be reused both for production and local testing
- Instructions on how to test

1. Visit an interaction partners page. For example: `/explore/Human-GEM/interaction-partners/ENSG00000069667`.
2. Click any node, which should slightly expand the sidebar height.
3. Verify that the labels do not shift. (For issue: #1176)
4. Upload data for `metabolites` in Data Overlay and click the "Add overlay" button.
5. Verify that the network viewer does not increase in height. (For issue: #1207)
6. Resize the window to mobile width.
7. Verify that the height adjusts accordingly.
8. Refresh the page while still on mobile width.
9. Verify that the viewer loads and is visible. (Item 2 under "List of changes made".)

**Further comments**  

Since the shifting in labels is related to the expandable network viewer height, this PR also targets #1207.

**Definition of Done checklist**  
- [x] My code meets the Acceptance Criteria
- [x] My code follows the [NBIS style guidelines](https://github.com/NBISweden/development-guidelines)
- [x] I have performed a self-review of my own code and commented any hard-to-understand areas
- [x] Tests and lint/format validations are passing
- [x] My changes generate no new warnings

### CODE REVIEW

**Review by nanjiangshu — COMMENTED** [review]
Elegant solution! It works well and great that you have solved the bug in the mobile view. 
I just have a minor comment that we need to upgrade the version of 3d-map-viewer in `frontend/package.json` so that Mac users can also click the node. 
```
+    "@metabolicatlas/3d-network-viewer": "^0.1.30",
```

File: frontend/src/components/explorer/interactionPartners/DetailsPage.vue [hunk 1]
```diff
@@ -537,7 +571,11 @@ export default {
               color = colorToRGBArray(this.defaultMetaboliteColor);
             }
           }
-          colors[node.id] = color;
+          if (node.id === this.clickedElmId) {
+            colors[node.id] = colorToRGBArray(NODE_SELECT_COLOR);
+          } else {
+            colors[node.id] = color;
+          }
```
Comment (inghylt) [inline frontend/src/components/explorer/interactionPartners/DetailsPage.vue:578]: Just curious, why is this included in the PR as it does not seem to be related to sizing?

Comment (e0) [reply frontend/src/components/explorer/interactionPartners/DetailsPage.vue:578]: Good question! In `handleWindowResize`, we need to call `await this.applyColorsAndRenderNetwork();`. If there is a selected node and we don't have this then the node would lose the red color. It's an edge case as I don't think users would be resizing the window a lot when using this page but I didn't want to introduce any new buggy behavior with this PR so it's included.

**Comment by e0** [comment]
> Elegant solution! It works well and great that you have solved the bug in the mobile view. I just have a minor comment that we need to upgrade the version of 3d-map-viewer in `frontend/package.json` so that Mac users can also click the node.
> 
> ```
> +    "@metabolicatlas/3d-network-viewer": "^0.1.30",
> ```

Thanks for the reminder @nanjiangshu! I think we can wait for https://github.com/MetabolicAtlas/3d-network-viewer/pull/64 to be merged and then I can add a new commit here that should include `0.1.31`.

File: frontend/src/components/explorer/interactionPartners/DetailsPage.vue (hunk 1 above)
Comment (inghylt) [reply frontend/src/components/explorer/interactionPartners/DetailsPage.vue:578]: Ah, makes sense! Nice catch

**Comment by e0** [comment]
> Seems to work very well, good job!
> 
> There is a warning from eslint though ([here](https://github.com/MetabolicAtlas/MetabolicAtlas/pull/1223/files#diff-f7d82ec59f2f3d52266fac6862a033ff93dc7846db3e51b8100d824f894cd7eaR77)).

Thank you! Fixed in: https://github.com/MetabolicAtlas/MetabolicAtlas/pull/1223/commits/46e4e415d158e2f41b2615889bed9e85a5226a9d

**Review by inghylt — APPROVED** [review]
Really nice fix! :star:

**Comment by e0** [comment]
Bumped dependency version in https://github.com/MetabolicAtlas/MetabolicAtlas/pull/1223 now

This is related to the following PRs:
- https://github.com/MetabolicAtlas/3d-network-viewer/pull/63
- https://github.com/MetabolicAtlas/3d-network-viewer/pull/64

**Pull request merged**